cmds/core/head/head.go

```diff
@@ -36,40 +36,58 @@ func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string)
 	var newLineHeader bool
 	var errs error
 
-	for _, file := range files {
-		f, err := os.Open(file)
-		if err != nil {
-			errs = errors.Join(errs, fmt.Errorf("head: %w", err))
-			continue
-		}
+	var handle = func(r io.Reader, name string) error {
 		if len(files) > 1 {
 			if newLineHeader {
-				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(f.Name()))
+				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(name))
 			} else {
-				fmt.Fprintf(stdout, "==> %s <==\n", filepath.Base(f.Name()))
+				fmt.Fprintf(stdout, "==> %s <==\n", filepath.Base(name))
 				newLineHeader = true
 			}
 		}
 		if printBytes {
-			n, err := io.ReadFull(f, buffer)
```

Comment:
We need to get read of this ReadFull. A named file can be a pipe, or a socket, or many things; or we could be reading a 10G file on a system with small memory (256MiB). We're going to need to use the buffo.NewScanner for every named file. Sorry.

Reply:
buffio.NewScanner is not really convenient because you need to deal with spaces, so I change buffer size to 4K

Reply:
See my note above. I was in a fog I guess.